pygfx/utils/text/_atlas.py

```diff
@@ -231,18 +255,26 @@ def _set_new_glyphs_array(self, size):
             info = self._infos[index]
             x1, y1 = info["origin"]
             w1, h1 = info["size"]
-            if not w1 or not h1:
-                continue  # freed region
+            # An index can have 0 size, for example the
+            # character 32, a non-printable character.
+            # has a glyph that is empty, but it should still
+            # take space and have an index in our atlas.
+            if index in self._free_indices:
+                # freed region
+                continue
             x2, y2, w2, h2 = self._select_region(w1, h1)
             info["origin"] = x2, y2
             array2[y2 : y2 + h2, x2 : x2 + w2] = array1[y1 : y1 + h1, x1 : x1 + w1]
-            allocated_area += w2 * h2
+            # 2 pixel padding for the allocated area
+            allocated_area += (w2 + 2) * (h2 + 2)
 
-        assert allocated_area == self._allocated_area
+        assert allocated_area == self._allocated_area, (
+            f"{allocated_area} != {self._allocated_area}"
+        )
 
         self._free_area = 0
         self._allocated_area = allocated_area
-        self._free_indices.clear()
+        # self._free_indices.clear()
```

Comment:
I actually think this used to be a bug, where _free_indices was never cleared. They were always just "there". and available for the taking since the `self._info` array isn't repacked.

Reply:
```suggestion
```
if you agree, you can press commit.